syntax = "proto3";

package input_proto;

message Params {
    string T_0 = 1;
    string T_1 = 2;
    double portfolio_risk_max = 3;
    double portfolio_return_min = 4;
    double portfolio_weight_max = 5;
    double portfolio_weight_min = 6;
    uint32 t = 7;
    double capital = 8;
    double weight_min = 9;
}

message InputData {
    string rqId = 1;
    string model = 2;
    string market = 3;
    string currency = 4;
    repeated string universe = 5;
    Params parameters = 6;
    int32 time_limit = 7;
    int32 iteration_limit = 8;
    int32 node_limit = 9;
    double user_obj_limit = 10;
    string topic_to_output = 11;
}